code/DynamicCache.php

```diff
@@ -384,6 +393,10 @@ public function run($url)
         // This prevents a new user's security token from being regenerated incorrectly
         $_SESSION['SecurityID'] = SecurityToken::getSecurityID();
 
+        // Set the stage of the website
+        // This is normally called in VersionedRequestFilter.
+        Versioned::choose_site_stage();
```

Comment:
Hm, this won't work without a current controller, not properly. See the code in VersionedRequestFilter.

Remember that all of this code runs before the main request chain, so request filter won't have a chance to bootstrap the application safely.

The best you can do is a direct access to the $_GET and $_SESSION vars to inspect for a current stage.